### PR TITLE
Members and Inventories migrations and association set up

### PR DESCRIPTION
**Pull request by skp96**
[Link to story](https://trello.com/c/A7HQ6Jvi/137-introducing-relationships-in-wine-o-inventory-app4)

[Link to ERD](https://drive.google.com/file/d/10ZagJaS4j8ZZQ0r7QsUdXpUUtdLneKjC/view?usp=sharing)

Migration Screenshot:
image

Highlights:

- Migrations for `Members` and `Inventories` tables
- `has_one` and `belongs_to` association between `Member` and `Inventory` respectively 
- added an `on_delete` option to `Member`'s `has_one` association as well as to the `Inventories` migration file per documentation
- `cast_assoc()` added to `Member`'s `changeset` function so when a `Member` is being created, Ecto can place instructions to create an associated `Inventory` record as well

### CODE REVIEW

File: lib/wine_o_inventory_api/inventories/inventory.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    belongs_to :member, WineOInventoryApi.Users.Member
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(inventory, attrs) do
+    inventory
+    |> cast(attrs, [])
+    |> validate_required([])
```
Comment (JeromeGoodrich) [inline lib/wine_o_inventory_api/inventories/inventory.ex]: If we aren't validating anything, is this required?

File: lib/wine_o_inventory_api/users/member.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+defmodule WineOInventoryApi.Users.Member do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "members" do
+    field :name, :string
+
+    has_one :inventory, WineOInventoryApi.Inventories.Inventory, on_delete: :delete_all
```
Comment (JeromeGoodrich) [inline lib/wine_o_inventory_api/users/member.ex:8]: based on some cursory reading it seems like this is not the suggested way of doing things https://hexdocs.pm/ecto/Ecto.Schema-macro-has_one.html#has_one/3-options and it's recommended to use `references` in the migration instead.

Comment (JeromeGoodrich) [reply lib/wine_o_inventory_api/users/member.ex:8]: I don't think you need both?

**Review by JeromeGoodrich — APPROVED** [review]

File: lib/wine_o_inventory_api/users/member.ex (hunk 2 above)
Comment (skp96) [reply lib/wine_o_inventory_api/users/member.ex:8]: I wasn't 100% sure which is why I left it in both places. I went through some posts on the elixir forum and some people added the option only in their migration file, while others left it in both places. My plan was to see if I run into any issues when writing tests to get a better understanding of how to exactly utilize this option.

File: lib/wine_o_inventory_api/inventories/inventory.ex (hunk 1 above)
Comment (skp96) [reply lib/wine_o_inventory_api/inventories/inventory.ex]: No, it is not! I'll go ahead and remove it.

**Review by k3ntako — APPROVED** [review]

**Pull request merged**